Python/pylifecycle.c

```diff
@@ -1848,8 +1855,15 @@ Py_FinalizeEx(void)
      * An address can be used to find the repr of the object, printed
      * above by _Py_PrintReferences.
      */
-    if (dump_refs) {
-        _Py_PrintReferenceAddresses(stderr);
+    if (dump_refs || dump_refs_file) {
+        if (fp == NULL) {
+            fprintf(stderr, "Can not log the addresses of all live objects.\n");
```

Comment:
I don't think that it's worth it to log two messages for the same error (fopen failure).